src/retrace/retrace.py

```diff
@@ -1310,6 +1310,59 @@ def package_name(self, debug=False):
     def needs_arch(self):
         return self._arch is None
 
+class KernelVMcore:
+    def __init__(self, vmcore_path):
+        self._vmcore_path = vmcore_path
+        self._is_flattened = None
+
+    def is_flattened_format(self):
+        """Returns True if vmcore is in makedumpfile flattened format"""
+        if not (self._is_flattened is None):
+            return self._is_flattened
+        try:
+            fd = open(self._vmcore_path)
+            fd.seek(0)
+            b = os.read(fd.fileno(), 512)
+        except OSError as e:
+            log_error("Failed to get makedumpfile header - failed open/seek/read of "
+                      "%s with errno(%d - '%s')" %
+                      (self._vmcore_path, e.errno, e.strerror()))
+        if fd:
+            fd.close()
+        match = re.match(b'makedumpfile', b)
```

Comment:
Actually, is there a reason why we need to read 512 byte if we only look at the first twelve?

Reply:
I wonder if there’s an actual description of the format. Is it generated in-kernel?

If there is a magic at the beginning, then it would make sense to not read as much, but, if not and if the layout is not deterministic, then doing this kind of makes sense.

Reply:
Oh, nevermind, I just read the Bugzilla bug. Yeah, we don’t need to read that much, just `SIG_LEN_MDF` bytes, AFAIU.

Reply:
It is checking for this string at the start of the file, which is the first portion of the header.  I didn't define the whole header which maybe isn't clear.  I chose 512 bytes because it was one sector and I would imagine in the future this function might be combined with other functions to determine which type of file this is.  For now I can change the size to be what is being read here.

Reply:
I guess if you read the crash-utility code, the header size is 4096 bytes.
https://github.com/crash-utility/crash/blob/master/makedumpfile.h#L31

SIG_LEN_MDF looks like 16 bytes to me.  What do you all want here?

Reply:
Going a bit off-topic here, but I’m thinking… Would it be possible to add the type to the file database (like https://github.com/file/file/blob/b67487c7fbcb8b5ec03dfe5d295137dd9ec37a39/magic/Magdir/linux#L486) and then simply use python-magic or some other wrapper for libmagic? Of course, we would need a new file release first.

Reply:
Other than that, I don’t really have a preference for the byte count that we read, so @mgrabovsky can decide.

Reply:
My preferences aren't strong either. If the magic length (`SIG_LEN_MDF`) is 16, let's go for that, as reading more seems unnecessary right now.